src/hotspot/share/runtime/javaThread.hpp

```diff
@@ -1136,6 +1135,7 @@ class JavaThread: public Thread {
   // clearing/querying jni attach status
   bool is_attaching_via_jni() const { return _jni_attach_state == _attaching_via_jni; }
   bool has_attached_via_jni() const { return is_attaching_via_jni() || _jni_attach_state == _attached_via_jni; }
+  inline void set_is_attaching_via_jni();
```

Comment:
So the only thing I don't like about this aspect is that this has to be called at construction time, so having it be a stand-alone function call invites misuse. I'm tempted to add a factory method:
```
JavaThread* JavaThread::new_attaching_thread() {
  JavaThread* jt = new JavaThread();
  jt>set_is_attaching_via_jni();
}
```
and make `set_is_attaching_via_jni()` private.

What do you think?

Reply:
Yes, I like that better, too. Okay, I'll do that.